[fix] Member도메인 리팩토링 및 테스트 코드 수정

ghrltjdtprbs opened the pull request:
- fcmember 관련 클래스 모두 삭제
- member 도메인 테스트 코드 수정 완료

ghrltjdtprbs committed:
[refactor] mypage 엔드포인트 재수정

문제 사항 :
 - mypage엔드포인트 수정 사항 반영 안됨
변경 사항 :
 - MemberController mypage url 재수정
(/api/v1/mypage   ->   /api/v1/mypages)

ghrltjdtprbs committed:
[refactor] fcmember관련 클래스 삭제

문제 사항 :
 - 더 이상 사용하지 않는 테이블
변경 사항 :
 - fcmember관련 클래스 모두 삭제

ghrltjdtprbs committed:
[fix] fcmember엔터티 삭제로 인한 에러해결

문제 사항 :
 - fcmember엔터티 삭제로 인해 일부 파일에 에러 발생
변경 사항 :
 - 에러 수정

ghrltjdtprbs committed:
Revert "[fix] fcmember엔터티 삭제로 인한 에러해결"

This reverts commit c577aa178af509fb88f4f1cf418addefd52139b5.

ghrltjdtprbs committed:
Revert "Revert "[fix] fcmember엔터티 삭제로 인한 에러해결""

This reverts commit efd03598869fcec72098e3ba5d7546b44e361160.

ghrltjdtprbs committed:
[fix] fcmember엔터티 삭제로 인한 에러해결

문제 사항 :
 - fcmember엔터티 삭제로 인해 일부 파일에 에러 발생
변경 사항 :
 - 에러 수정

ghrltjdtprbs committed:
[fix] 회원정보 조회, 수정 docs test 수정

문제 사항 :
 - jwt 인증방식으로 변경해 회원정보 조회, 수정 테스트 실패
변경 사항 :
 - 에러 수정

ghrltjdtprbs committed:
[fix] 회원정보 조회, 수정 docs test 수정

문제 사항 :
 - jwt 인증방식으로 변경해 회원정보 조회, 수정 테스트 실패
변경 사항 :
 - 에러 수정

ghrltjdtprbs committed:
[fix] MemberControllerTest 수정

변경 사항 :
 - jwt인증방식 변경으로 인해 실패하는 테스트 코드 수정
 - 회원 탈퇴, 마이페이지 조회 로직 수정

chimaek commented on src/main/java/com/fasttime/domain/member/service/MemberService.java:
# 1. registerOrRecoverMember
두 가지 기능(등록과 복구)을 가지고있네요, 이러한 메소드는 각각의 기능별로 분리하는 것이 좋습니다. 

# 2. save 메서드 두 개 존재
save 메소드가 두 개 있는데요, 하나는 CreateMemberDTO를 받고 다른 하나는 Member를 받고있죠? 더 명확하게 구분하기 위해 saveNewMember와 updateMember와 같은 이름을 사용해보시는건 어떨까요?

# 3. DTO 이름 통일
EditRequest와 CreateMemberDTO을 사용하고 있는데, 왠만하면 클래스이름또한 비슷하게 맞춰가시길 추천드립니다 :)

ghrltjdtprbs replied:
피드백 감사합니다!! 전부 반영해 리팩토링 진행하였습니다. 새로운 커밋 확인해주세요!😊😊

chimaek reviewed (commented):
고생하셨습니당 👍

ghrltjdtprbs committed:
[refactor] Member 도메인 DTO이름 통일

변경 사항 :
 - 통일성을 위해 member도메인의 dto이름을 변경하였습니다.

ghrltjdtprbs committed:
[refactor] MemberService save메서드명 변경

문제 사항 :
 - MemberService내에 save 메서드 두 개 존재
변경 사항 :
 - 명확한 구분을 위해 처음 회원가입시 이용하는 save 메서드명을 saveNewMember로 변경
 - 복구 로직에 사용되는 save 메서드 삭제

ghrltjdtprbs committed:
[refactor] 회원 가입, 복구 기능 분리

문제 사항 :
 - registerOrRecoverMember 메서드에 등록과 복구 로직 동시에 존재
변경 사항 :
 - 등록과 복구 로직 분리

chimaek reviewed (approved):
합격

JeongUijeong reviewed (approved):
수고하셨습니다 👍🏻👍🏻

Nine-JH reviewed (approved):
고생하셨어요 API 경로 변경건에 관해서는 프론트엔드 분들에게도 알려주세요!
